src/lakefs_spec/client_helpers.py

```diff
@@ -189,11 +185,29 @@ def create_tag(
     try:
         return client.tags_api.create_tag(repository=repository, tag_creation=tag_creation)
     except ApiException as e:
-        if e.status == 409 and exist_ok:
-            return client.tags_api.get_tag(repository=repository, tag=tag)
+        target_commit = rev_parse(client=client, repository=repository, ref=ref)
+        existing_tag = client.tags_api.get_tag(repository=repository, tag=tag)
+        if e.status == 409 and existing_tag.commit_id == target_commit.id:
+            return existing_tag
```

Comment:
For performance reasons, please move the rev-parse attempt in the `if e.status == 409` block so that it doesn't execute on e.g. a `PermissionError`:

```suggestion
        if e.status == 409:
                tagged_commit = rev_parse(client=client, repository=repository, ref=ref)
                existing_tag = client.tags_api.get_tag(repository=repository, tag=tag)
                if tagged_commit.id == existing_tag.commit_id:
                    return existing_tag
```